cmd/kpod/pull.go

```diff
@@ -132,31 +65,34 @@ func pullCmd(c *cli.Context) error {
 		return err
 	}
 	image := args[0]
-	srcRef, err := alltransports.ParseImageName(image)
-	if err != nil {
-		fqRegistries, err = getRegistriesToTry(image)
+
+	var registryCreds *types.DockerAuthConfig
+	if c.String("creds") != "" {
+		creds, err := common.ParseRegistryCreds(c.String("creds"))
 		if err != nil {
-			fmt.Println(err)
+			if err == common.ErrNoPassword {
+				fmt.Print("Password: ")
+				password, err := terminal.ReadPassword(0)
+				if err != nil {
+					return errors.Wrapf(err, "could not read password from terminal")
+				}
+				creds.Password = string(password)
+			} else {
+				return err
+			}
 		}
-	} else {
-		fqRegistries = append(fqRegistries, srcRef.DockerReference().String())
+		registryCreds = creds
 	}
-	runtime, err := getRuntime(c)
-	if err != nil {
-		return errors.Wrapf(err, "could not get runtime")
-	}
-	defer runtime.Shutdown(false)
 
-	if err != nil {
-		return errors.Wrapf(err, "could not create runtime")
-	}
-	for _, fqname := range fqRegistries {
-		fmt.Printf("Trying to pull %s...", fqname)
-		if err := runtime.PullImage(fqname, c.Bool("all-tags"), c.String("signature-policy"), os.Stdout); err != nil {
-			fmt.Printf(" Failed\n")
-		} else {
-			return nil
-		}
+	options := libpod.CopyOptions{
+		SignaturePolicyPath: c.String("signature-policy"),
+		AuthFile:            c.String("authfile"),
+		DockerRegistryOptions: common.DockerRegistryOptions{
+			DockerRegistryCreds: registryCreds,
+		},
+		Writer: os.Stdout,
```

Comment:
I think we should have a --quiet option for pull.  If we do, we'll need to handle Writer like we do in push.

Reply:
Any thoughts on a quiet option?

Reply:
Doesn't have to hold this submit, I'd be happy with an issue/trello card if we're going to do it.